### PR TITLE
add updated preprocessing and tests

### PR DESCRIPTION
**Pull request by cameronmartino**
### Change log:
- Directly use fastqs to build graphs via RBFG
- How graphs are built
    1. specify kmer length (default is 21)
    2. if fit is used all leave-out-out edges and all conditional graphs are build 
    3. Each conditional graph has n red edges for everything in condition and at most one blue edge for anything in the leave-one-out graph that is not in the condition 

### Example of use 
```python 
from TesasSemble.preprocessing import RBFG
``` 
Example fastqs in tests/data/fastq_test
    - a fist sample S0.fastq
    - a second sample S1.fastq

```python 
g1 = 'tests/data/fastq_test/S0.fastq'
g2 = 'tests/data/fastq_test/S1.fastq'
G_RdBu = RBFG(k=21).fit([g1,g2])
```   
Now we can get the conditional graphs for each.
  
```python   
G1_RdBu  = G_RdBu.condition_graph(g1)
G2_RdBu  = G_RdBu.condition_graph(g2)
``` 
Now we can also map each node back to it's kmer.

```python     
node_fasta = G_RdBu.node_map()

``` 
Alternatively, you can fit_transform for one condition. You will need to pass a map of the nodes at each iteration. This will be faster if you only need one of the conditions but slightly slower than doing them all at once. So if you need all of the conditions it is recommended to run ```  RBFG(k=12).fit([g1,g2,...,g_n]) ``` .

```python     
G1_RdBu_single, node_map = RBFG().fit_single([g1,g2], g1)
G2_RdBu_single, node_map = RBFG().fit_single([g1,g2], g2, nodes = node_map)
```

### CODE REVIEW

File: TesasSemble/preprocessing.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
+import numpy as np
+from Bio import SeqIO
+from TesasSemble.graph import RedBlueDiGraph
+from TesasSemble.graph_components import Node, Edge
+from TesasSemble.base import _BaseRedBlueFastq
+
+
+class RBFG(_BaseRedBlueFastq):
```
Comment (gwarmstrong) [inline TesasSemble/preprocessing.py]: Can we give this a slightly more informative name?

File: TesasSemble/preprocessing.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        GRdBu.add_edges_from([edge_G_tmp[str(n[0]) + '-' + str(n[1])]
+                              for n in blue],
+                             ['blue'] * len(blue))
+        self.gtmp = GRdBu.copy()
+        return
+
+
+def debruijn(seqs, k):
+    # debruijn helper
+    kmers = []
```
Comment (gwarmstrong) [inline TesasSemble/preprocessing.py]: ``` kmers ``` should really be called ```edges``` and ```seqs``` should be called ```kmers```

File: TesasSemble/preprocessing.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+    def fit(self, conditions):
+
+        self.conditions = conditions
+
+        # generate edges
+        self.edges = {r: debruijn([str(fasta.seq) for fasta
+                                   in SeqIO.parse(open(r),
+                                                  'fasta')], self.k)
+                      for r in self.conditions}
```
Comment (gwarmstrong) [inline TesasSemble/preprocessing.py]: Call ```r``` something more informative like ```condition```

**Review by gwarmstrong — APPROVED** [review]

File: TesasSemble/preprocessing.py (hunk 1 above)
Comment (cameronmartino) [reply TesasSemble/preprocessing.py]: @gwarmstrong Are you referring to the name RBFG? , just making sure.

Comment (gwarmstrong) [reply TesasSemble/preprocessing.py]: @cameronmartino Yes, I am referring to RBFG

Comment (cameronmartino) [reply TesasSemble/preprocessing.py]: @gwarmstrong  Any suggestions?

Comment (gwarmstrong) [reply TesasSemble/preprocessing.py]: ``` GraphConstructor ```?

**Review by gwarmstrong — APPROVED** [review]

**Pull request merged**